logstash-core/spec/logstash/api/commands/stats_spec.rb

```diff
@@ -35,18 +73,93 @@
 
   describe "memory stats" do
     let(:report_method) { :memory }
-      
+
     it "return hot threads information" do
       expect(report).not_to be_empty
     end
 
-    it "return heap information" do
-      expect(report.keys).to include(:heap_used_in_bytes)
+    it "return memory information" do
+      expect(report.keys).to include(
+        :heap_used_percent,
+        :heap_committed_in_bytes,
+        :heap_max_in_bytes,
+        :heap_used_in_bytes,
+        :non_heap_used_in_bytes,
+        :non_heap_committed_in_bytes,
+        :pools
+      )
     end
+  end
 
-    it "return non heap information" do
-      expect(report.keys).to include(:non_heap_used_in_bytes)
+  describe "jvm stats" do
+    let(:report_method) { :jvm }
+
+    it "return jvm information" do
+      expect(report.keys).to include(
+        :threads,
+        :mem,
+        :gc,
+        :uptime_in_millis
+      )
+      expect(report[:threads].keys).to include(
+        :count,
+        :peak_count
+      )
     end
+  end
 
+  describe "reloads stats" do
+    let(:report_method) { :reloads }
+
+    it "return reloads information" do
+      expect(report.keys).to include(
+      :successes,
+      :failures,
+      )
+    end
   end
+
+  describe "pipeline stats" do
+    let(:report_method) { :pipeline }
+
+    it "returns pipelines information" do
```

Comment:
I suggest breaking this large test into smaller checks, e.g:

```ruby
    it "returns information on existing pipeline" do
      expect(report.keys).to include(:main)
    end
    context "for each pipeline" do
      it "returns information on pipeline" do
        expect(report[:main].keys).to include(
          :events,
          :plugins,
          :reloads,
          :queue,
        )
      end
      it "returns event information" do
        expect(report[:main][:events].keys).to include(
          :in,
          :filtered,
          :duration_in_millis,
          :out,
          :queue_push_duration_in_millis
        )
      end
    end
```

Reply:
Thanks. Suggestion applied.